c7n/resources/ec2.py

```diff
@@ -206,6 +206,34 @@ def get_resource_date(self, i):
         return None
 
 
+class ASGFilter(object):
+    """Filter instances by whether they are members of an ASG"""
+    asg_member = None
+
+    def filter_asg_membership(self, instances, desired=None):
+        if desired is None:
+            desired = self.asg_member
+
+        if self.asg_member is None:
+            # TODO raise exception
```

Comment:
its unclear that this filter should even exist. its already available as a value filter on the tag presence. where we have existing filters that provide the same capability there's a strong preference to use those vs adding in one-offs.

ie. the equivalent value filter
```
type: value
key: "tag:aws:autoscaling:groupName"
value: not-null
```